src-front/src/components/WordCloud/WordCloud.tsx

```diff
@@ -154,17 +154,38 @@ export default function WordCloud({
   const [inputValue, setInputValue] = useState("");
   //const [error, setError] = useState("");
   const [scores, setScores] = useState<Record<string, number>>({});
-  const [wordsBestProgress, setWordsBestProgress] = useState<
-    Record<string, number>
-  >({});
+  // const [wordsBestProgress, setWordsBestProgress] = useState<
+  //   Record<string, number>
+  // >({});
   const [gameEndEpochMs, setGameEndEpochMs] = useState(0);
   const [gameStartEpochMs, setGameStartEpochMs] = useState(0);
   const fightSession = useRef<FightSession | null>(null);
   const refAddWord = useRef<(wordStr: string) => void>();
   const refDeleteWord = useRef<(wordToDelete: string) => void>();
+  const refPreviousEntry = useRef<string>("");
 
   //WordCloud initialization
   const init = useCallback(async () => {
+    //forbids pasting text
+    const playerInput = document.getElementById(
+      "playerInput",
+    ) as HTMLInputElement;
+    playerInput.onpaste = (e) => e.preventDefault();
+    //forbids text selection
+    playerInput.addEventListener(
```

Comment:
It needs an equivalent remove event listener for each event listener created.

Reply:
I just removed it, it wasn't a good idea anyway

Reply:
But I mean, for all events.